newIDE/app/src/InstancesEditor/FullSizeInstancesEditorWithScrollbars.js

```diff
@@ -33,154 +76,412 @@ type Props = {|
   wrappedEditorRef: ?(?InstancesEditor) => void,
 |};
 
-type State = {|
-  xValue: number,
-  yValue: number,
-  xMin: number,
-  xMax: number,
-  yMin: number,
-  yMax: number,
-|};
-
 const noop = () => {};
 
-export default class FullSizeInstancesEditorWithScrollbars extends Component<
-  Props,
-  State
-> {
-  _editor: ?InstancesEditor = null;
-
-  state: State = {
-    xValue: 0,
-    yValue: 0,
-    xMin: -5000,
-    xMax: +5000,
-    yMin: -5000,
-    yMax: +5000,
-  };
-
-  componentDidMount() {
-    if (this._editor) {
-      this._handleViewPositionChange(this._editor.getViewPosition());
-    }
-  }
-
-  _handleXChange = (e: any, value: number) => {
-    const { xMin, xMax } = this.state;
-    const xValue = xMin + ((xMax - xMin) * value) / 100;
-    this.setState(
-      {
-        xValue,
-      },
-      () => {
-        if (this._editor) {
-          this._editor.scrollTo(this.state.xValue, this.state.yValue);
-        }
+const FullSizeInstancesEditorWithScrollbars = (props: Props) => {
+  const { wrappedEditorRef, ...otherProps } = props;
+
+  const editorRef = React.useRef<?InstancesEditor>(null);
+  const xScrollbarDetectionZone = React.useRef<?HTMLDivElement>(null);
+  const xScrollbarTrack = React.useRef<?HTMLDivElement>(null);
+  const xScrollbarThumb = React.useRef<?HTMLDivElement>(null);
+  const yScrollbarDetectionZone = React.useRef<?HTMLDivElement>(null);
+  const yScrollbarTrack = React.useRef<?HTMLDivElement>(null);
+  const yScrollbarThumb = React.useRef<?HTMLDivElement>(null);
+
+  const showScrollbars = React.useRef(false);
+  const timeoutHidingScrollbarsId = React.useRef<?TimeoutID>(null);
+  const isDragging = React.useRef(false);
+
+  const xValue = React.useRef(0);
```

Comment:
xScrollValue and yScrollValue ?

Reply:
well technicaly it's not related to scroll, it defines where the viewPosition is positioned